sycl/include/CL/sycl/backend.hpp

```diff
@@ -49,7 +49,7 @@ template <backend Backend> class backend_traits {
   template <class T>
   using return_type = typename detail::BackendReturn<Backend, T>::type;
 
-  // TODO define errc once SYCL2020-style exceptions are supported.
+  using errc = sycl::errc;
```

Comment:
I would expect that we should have a separate `enum` with values unique to OpenCL BE.
Could you please clarify why the patch makes it an alias of `sycl::errc`?

Reply:
These isn't the OpenCL backend, just the default one.  If OpenCL needs its own values then we would specialize it. I was focused on just getting in the minimal support for `errc_for`  .  Are there different values for OpenCL?

Reply:
I'm not completely sure, but an example in `4.13.1.4. Asynchronous errors with a secondary queue` suggests that the enum should contain backend specific errors(ala CL_INVALID_VALUE). Also there is ` C.7.4. Errors and limitations` which suggest that there should be a special function for extracting "native" value, however this is only mention of this function in the spec.
P.S. I was looking at the Spec from SYCL-2020/final-rev3 commit: 3aa000fbe94ab385edb02d24cadc0b94acd45604 . Maybe all these things are clarified on the top of the trunk.

If the main reason of this patch is to just a type, I would suggest defining a new empty enum. A new because if we currently map it to `sycl::errc` and later decide to create a separate one it would break backward compatibility.

Tagging @gmlueck

Reply:
The idea is that each backend could have a separate enum of error codes that is different from `sycl::errc`.  I agree with @romanovvlad that this should **not** be an alias to `sycl::errc`.

The SYCL spec should be improved in this area, and I opened an issue against the spec to capture this problem:

https://github.com/KhronosGroup/SYCL-Docs/issues/182

In the mean time, I suggest either leaving the TODO comment as it was before or defining an empty OpenCL-specific enumeration.

Reply:
Thanks @gmlueck .  I put an empty enumeration with a TODO comment where the previous had been.